assignment/impl/src/java/org/sakaiproject/assignment/impl/AssignmentServiceImpl.java

```diff
@@ -1403,32 +1403,58 @@ public void updateAssignment(Assignment assignment) throws PermissionException {
             throw new PermissionException(sessionManager.getCurrentSessionUserId(), SECURE_UPDATE_ASSIGNMENT, null);
         }
 
-        Collection<String> oldGroups = assignmentRepository.findGroupsForAssignmentById(assignment.getId());
+        Collection<String> prevAssignedGroupRefs = assignmentRepository.findGroupsForAssignmentById(assignment.getId());
         switch (assignment.getTypeOfAccess()) {
             case GROUP:
-                oldGroups.removeAll(assignment.getGroups());
-                for (String groupRef : oldGroups) { // remove locks for groups that were removed
-                    try {
-                        AuthzGroup group = authzGroupService.getAuthzGroup(groupRef);
-                        group.setLockForReference(reference, AuthzGroup.RealmLockMode.NONE);
-                        authzGroupService.save(group);
-                    } catch (GroupNotDefinedException | AuthzPermissionException e) {
-                        log.warn("Exception while removing lock for assignment {}, {}", assignment.getId(), e.toString());
+                Collection<String> currentAssignedGroupRefs = assignment.getGroups();
+                // get the unassigned Group References by removing the current ones !
+                prevAssignedGroupRefs.removeAll(currentAssignedGroupRefs);
+
+                Site site = null;
+                try {
+                    site = siteService.getSite(assignment.getContext());
+                } catch (IdUnusedException e) {
+                    log.warn("Could not get the site {} for assignment {} while updating it", assignment.getContext(), assignment.getId());
+                }
+                // on Lessons, a group assignment or one with prereq. will be managed through a single (access) group
+                Group anAssignedGroup = site.getGroup(currentAssignedGroupRefs.stream().findFirst().orElse(null));
```

Comment:
I don't get this, why do you only check the first group ? Is this logic called through assignments or through lessons?

And why do you get the Group from the site? Because the AuthzGroup does not have properties ?

Reply:
If you assign 3 groups to an assignment (when creating or updating it), `currentAssignedGroupRefs` will contain three references. It's ok to check just one, none of them will have the "lessonbuilder_ref" prop.

If later you follow the testplan on Lessons, especially when setting the assignment to don't be released until prereqs are completed - `prevAssignedGroupRefs` will now contain the three group refs, and `currentAssignedGroupRefs` will only reference the new ACCESS group, which is used to manage the user visibility of the item on Lessons, an has the prop.

Right I couldn't get it through AuthzGroup

Reply:
Okay thanks